src/Interpreters/OpenTelemetrySpanLog.cpp

```diff
@@ -101,6 +101,14 @@ OpenTelemetrySpanHolder::~OpenTelemetrySpanHolder()
 {
     try
     {
+        // If span is not ready we don't try to save it.
+        // It may be broken by bad_alloc when filled or similar
+        if (!is_ready)
+        {
+            assert(std::uncaught_exceptions());
+            return;
```

Comment:
I think this `return` should be put after line 110 in original file where the context on current thread is restored.